waf_test.go

```diff
@@ -253,201 +250,9 @@ func TestActions(t *testing.T) {
-				},
-			},
-		}
-
-		testEmptyRulesData = []rc.ASMDataRuleData{
-			{
-				ID:   "blocked_users",
-				Type: "data_with_expiration",
-				Data: []rc.ASMDataRuleDataEntry{},
-			},
-			{
-				ID:   "blocked_ips",
-				Type: "ip_with_expiration",
-				Data: []rc.ASMDataRuleDataEntry{},
-			},
-		}
-	)
-
-	waf, err := newDefaultHandle(testBlockingRule)
-	require.NoError(t, err)
-	require.NotNil(t, waf)
-	defer waf.Close()
-
-	// Helper function to test that the given address blocks or not.
-	// A rule can still only match once per context and so this function helps
-	// testing several times the same rule under distinct rule data values.
-	test := func(values map[string]interface{}, expectedActions []string) {
-		wafCtx := NewContext(waf)
-		require.NotNil(t, wafCtx)
-		defer wafCtx.Close()
-
-		matches, actions, err := wafCtx.Run(values, time.Second)
-		require.NoError(t, err)
-
-		if len(expectedActions) > 0 {
-			require.NotEmpty(t, matches)
-			require.Equal(t, len(expectedActions), len(actions))
-			require.ElementsMatch(t, expectedActions, actions)
-		} else {
-			require.Nil(t, matches)
-			require.Nil(t, actions)
-		}
-	}
-
-	// Not matching because the address values don't match the rules data
-	test(map[string]interface{}{
-		"http.client_ip": "10.0.0.1",
-		"usr.id":         "moutix",
-	}, nil)
-
-	// Matching because the address values match the rules data
-	test(map[string]interface{}{
-		"http.client_ip": "1.2.3.4",
-		"usr.id":         "zouzou",
-	}, []string{"block_user", "block_ip"})
-
-	// Update the rules' data
-	err = waf.UpdateRulesData(testBlockingRulesData)
-	require.NoError(t, err)
-
-	// Not matching because the address values match the updated rules data
-	test(map[string]interface{}{
-		"http.client_ip": "1.2.3.4",
-		"usr.id":         "zouzou",
-	}, nil)
-
-	// Matching because the address values don't match the updated rules data
-	test(map[string]interface{}{
-		"http.client_ip": "10.0.0.1",
-		"usr.id":         "moutix",
-	}, []string{"block_user", "block_ip"})
-
-	// Empty the rules data so that nothing matches anymore
-	err = waf.UpdateRulesData(testEmptyRulesData)
-	require.NoError(t, err)
-
-	test(map[string]interface{}{
-		"http.client_ip": "1.2.3.4",
-		"usr.id":         "zouzou",
-	}, nil)
-
-	test(map[string]interface{}{
-		"http.client_ip": "10.0.0.1",
-		"usr.id":         "moutix",
-	}, nil)
-
-	// Update the rules' data again
-	err = waf.UpdateRulesData(testBlockingRulesData)
-	require.NoError(t, err)
-
-	// Matching because the address values don't match the updated rules data
-	test(map[string]interface{}{
-		"http.client_ip": "10.0.0.1",
-		"usr.id":         "moutix",
-	}, []string{"block_user", "block_ip"})
-}
-
 func TestAddresses(t *testing.T) {
 	defer requireZeroNBLiveCObjects(t)
-	expectedAddresses := []string{"my.first.input", "my.second.input", "my.third.input", "my.indexed.input"}
+	expectedAddresses := []string{"my.first.input", "my.second.input", "my.indexed.input", "my.third.input"}
```

Comment:
Not sure I understand this one. Shouldn't `my.third.input` be third?

Reply:
It's the new output of libddwaf v1.8
cc @Anilm3 if you want to have a look, but I am okay with it.

Reply:
Makes sens, we'll adjust the test next time it breaks, thanks.